---
title: "Functional epigenetic clocks reveal system-discordant tissue aging in individuals with cancer"
author: "Supplementary Information"
date: ""
link-citations: true
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
header-includes:
  \usepackage{fontspec}
  \usepackage[utf8]{inputenc}
  \setmainfont{Helvetica}
  \usepackage{pdfpages}
  \usepackage{graphicx}
  \newfontfamily\boldf{Helvetica Bold}
---

Chiara Herzog, Elisa Redl, James Barrett, Sepideh Aminzadeh-Gohari, Daniela Weber, Julia Tevini, Roland Lang, Barbara Kofler, and Martin Widschwendter

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, message = F, warning = F)

# libraries
library(ggplot2)
library(dplyr)
library(ComplexHeatmap)
library(RColorBrewer)
library(ggtext)
library(patchwork)
library(here)

# colours
cols <- MetBrewer::met.brewer("Hiroshige", n = 12)
cols1 <- MetBrewer::met.brewer("Hokusai1", n = 8)
```

\newpage

\tableofcontents

\newcommand\invisiblesection[1]{%
  \refstepcounter{section}%
  \addcontentsline{toc}{section}{\protect\numberline{\thesection}#1}%
  \sectionmark{#1}}

\newpage

\invisiblesection{Figure S1. Identification of senescence-associated CpGs.}

```{r fig.s1.senescence, fig.width = 10, fig.height = 14, eval = F}
load(here("1-functional-association/1-senescence/3-output/coef.Rdata"))

a <- df |>
  ggplot(aes(x = p)) +
  geom_histogram(bins = 300) +
  theme_minimal() +
  labs(x = "p value", y = "count")

b <- df |>
  ggplot(aes(x = coef)) +
  geom_histogram(bins = 300) +
  theme_minimal() +
  xlim(c(-0.3, 0.3)) +
  labs(x = "coefficient (senescence association)", y = "count")

load(here("1-functional-association/1-senescence/4-output/pheno_senescence_association.Rdata"))

c <- pheno |>
  tibble::rownames_to_column("id") |>
  dplyr::mutate(type = ifelse(type == "control", "Control",
                              "Senescent cells"),
                ct = stringr::str_wrap(celltype, width = 30),
                induction = stringr::str_wrap(paste0(induction, " senescence"), width = 30)) |>
  ggplot(aes(x = type,
             y = idx,
             fill = type)) +
  geom_boxplot() +
  ggbeeswarm::geom_quasirandom() +
  facet_wrap(ct ~ induction,
             nrow = 1) +
  labs(x = "", y = "weighted senescence CpG methylation") +
  theme_bw() +
  theme(legend.position = "top",
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  scale_fill_manual(values = cols[c(5, 1)],
                    name = "Type") +
  ggpubr::stat_compare_means(aes(label = paste0('p = ', after_stat('p.format'))),
                             comparisons = list(c("Control", "Senescent cells")),
                             size = 3,
                             method = 'wilcox')

load(here("1-functional-association/1-senescence/5-output/pheno.Rdata"))
t <- pheno |>
  dplyr::group_by(tissue_simplified) |>
  dplyr::count() |>
  dplyr::filter(n > 3)

d <- pheno |>
  dplyr::filter(tissue_simplified %in% t$tissue_simplified) |>
  ggplot(aes(x = idx,
             y = log2(p16+0.001),
             colour = tissue_simplified)) +
  geom_point(size = 0.75) +
  geom_smooth(method = "lm",
              se = F) +
  ggpubr::stat_cor(label.y.npc = "bottom",
                   label.x.npc = "middle") +
  theme_bw() +
  theme(axis.title.y = element_markdown()) +
  scale_color_manual(values = colorRampPalette(cols1)(13), name = "tissue") +
  labs(x = "weighted senescence CpG methylation", y = "log2(<i>CDKN2A</i> mRNA expression)")

plot <- ((a|b)/c/(d+plot_spacer())) + plot_annotation(tag_levels = 'A')
print(plot)

ggsave(plot, file = here("4-display-elements/panels/figs1.pdf"),
       width = 10, height = 14)
```

\begin{center}
\includegraphics[trim=0 0cm 0 0, clip,width=0.7\textwidth]{./panels/figs1.pdf}
\end{center}

\begingroup \boldf Figure S1. Identification of senescence-associated CpGs. A \endgroup Histogram of p values for differential methylation analysis of senescence status, accounting for dataset and cell type. \textbf{B} Coefficient for association with methylation value by time reveals more loss than gain in methylation with senescence. \textbf{C} The weighted senescence methylation score is elevated in all senescence datasets used for development of the signature. Boxplot indicates median and interquartile range with overlaid individual data points. P values were computed comparing control and senescent cells using a two-sided Wilcoxon test. \textbf{D} The weighted senescence methylation score is positively correlated with mRNA expression of the key senescence-associated gene \textit{CDKN2A} (p16) in 8/13 evaluated normal tissues with matched methylation and mRNA expression data from TCGA (individual data points are plotted). Correlation coefficient is Pearson's rho.

\newpage

\invisiblesection{Figure S2. Identification of proliferation-associated CpGs.}

```{r fig.s2.proliferation, fig.width = 10, fig.height = 14, eval = F}
load(here("1-functional-association/2-proliferation/3-output/coef.Rdata"))

a <- df |>
  ggplot(aes(x = p)) +
  geom_histogram(bins = 300) +
  theme_minimal() +
  labs(x = "p value", y = "count")

b <- df |>
  ggplot(aes(x = coef)) +
  geom_histogram(bins = 300) +
  theme_minimal() +
  xlim(c(-0.3, 0.3)) +
  labs(x = "coefficient (proliferation association)", y = "count")

load(here("1-functional-association/2-proliferation/4-output/pheno_association.Rdata"))

c <- pheno |>
  dplyr::mutate(subexperiment = ifelse(subexperiment == "MMC", "mitomycin C",
                                       "serum withdrawal"),
                tissue = tolower(tissue)) |>
  ggplot(aes(x = as.numeric(days_in_culture),
             y = idx_inverted,
             colour = type)) +
  geom_point() +
  facet_wrap(tissue~subexperiment) +
  labs(x = "days in culture", y = "weighted proliferation CpG methylation") +
  theme_bw() +
  theme(legend.position = "top") +
  scale_colour_manual(values = cols[c(8,2)],
                    name = "proliferation blocker")

load(here("1-functional-association/2-proliferation/5-output/pheno.Rdata"))
t <- pheno |>
  dplyr::group_by(tissue_simplified) |>
  dplyr::count() |>
  dplyr::filter(n > 3)

d <- pheno |>
  dplyr::filter(tissue_simplified %in% t$tissue_simplified) |>
  ggplot(aes(x = idx,
             y = log2(ki67+0.001),
             colour = tissue_simplified)) +
  geom_point(size = 0.75) +
  geom_smooth(method = "lm",
              se = F) +
  ggpubr::stat_cor(label.y.npc = "bottom",
                   label.x.npc = "middle") +
  theme_bw() +
  theme(axis.title.y = element_markdown()) +
  scale_color_manual(values = colorRampPalette(cols1)(13), name = "tissue") +
  labs(x = "weighted proliferation CpG methylation", y = "log2(<i>MK67</i> mRNA expression)")

plot <- ((a|b)/c/(d+plot_spacer())) + plot_annotation(tag_levels = 'A')
print(plot)

ggsave(plot, file = here("4-display-elements/panels/figs2.pdf"),
       width = 10, height = 14)
```

\begin{center}
\includegraphics[trim=0 0cm 0 0, clip,width=0.8\textwidth]{./panels/figs2.pdf}
\end{center}

\begingroup \boldf Figure S2. Identification of proliferation-associated CpGs. A \endgroup Histogram of p values for differential methylation analysis of (reduced) proliferation status, accounting for tissue type and subexperiment (mitomycin C or serum withdrawal). \textbf{B} Coefficient for association with methylation value by time reveals more gain than loss of methylation with reduced proliferation \textbf{C} The weighted proliferation methylation score is consistently reduced with reduced proliferation across subexperiments in the dataset used to identify sites. \textbf{D} The weighted proliferation methylation score is positively correlated with mRNA expression of the key proliferation-associated gene \textit{MKI67} (ki67) in 9/13 evaluated normal tissues with matched methylation and mRNA expression data from TCGA. Correlation coefficient is Pearson's rho.

\newpage

\invisiblesection{Figure S3. Association of new clocks with age.}

```{r fig.s3, eval = F}
load(here("3-evaluation/3-output/data.Rdata"))

tmp <- data |>
  dplyr::filter(group == "current cancer;surrogate tissue" & type == "Control") |>
  dplyr::mutate(icgrp = ifelse(celltype %in% c("cervical sample") & ic < 0.2, "epithelial fraction",
                               ifelse(celltype == "blood sample", "immune fraction", NA))) |>
  dplyr::filter(!is.na(icgrp)) |>
  tidyr::pivot_longer(idx_AGE:phenoage,
                      names_to = "index",
                      values_to = "value") |>
  dplyr::mutate(index = gsub("idx_", "", index),
                index = gsub("_neg", "-", index),
                index = gsub("_pos", "+", index))

plot <- tmp |>
  ggplot(aes(x = age,
             y = value)) +
  geom_point(size = 0.55) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(icgrp ~ index,
             scales = "free_y") +
  ggpubr::stat_cor(size = 2.2,
                   colour = "blue") +
  theme_bw() +
  labs(x = "age",
       y = "score")

ggsave(plot, file = here("4-display-elements/panels/figs3.pdf"),
       width = 12, height = 14)
```

\begin{center}
\includegraphics[trim=0 0cm 0 0, clip,width=0.9\textwidth]{./panels/figs3.pdf}
\end{center}

\begingroup \boldf Figure S3. Correlation of clocks with age. \endgroup Clocks were correlated with age in epithelial cells (cervical or buccal samples with an inferred immune cell proportion of less that 20\%) or immune cells (blood samples) used to derive the signatures. Correlation coefficient is Pearson's rho.

\newpage

\invisiblesection{Figure S4. Representative examples of CpG sites passing cell type-specific criteria.}

```{r fig.s4.age, fig.width = 4.75, fig.height = 9, eval = F}
load(here("2-age-association/3-output/pheno_examples.Rdata"))
pheno2 <- pheno2 |>
  dplyr::mutate(icgrp = ifelse(icgrp == "epithelial", "epithelial fraction", "immune fraction"))

a <- pheno2 |>
  ggplot(aes(x = age,
             y = general_cg05024939)) +
  geom_point(size = 0.75) +
  geom_smooth(method = "lm") +
  facet_wrap(~ icgrp) +
  ggpubr::stat_cor() +
  theme_bw() +
  theme(aspect.ratio = 1) +
  labs(y = "cg05024939",
       title = "Shared (general) age-related CpG")

b <- pheno2 |>
  ggplot(aes(x = age,
             y = epi_cg11528849)) +
  geom_point(size = 0.75) +
  geom_smooth(method = "lm") +
  facet_wrap(~ icgrp) +
  ggpubr::stat_cor() +
  theme_bw() +
  theme(aspect.ratio = 1) +
  labs(y = "cg11528849",
       title = "Epithelial age-related CpG example")

c <- pheno2 |>
  ggplot(aes(x = age,
             y = imm_cg20786223)) +
  geom_point(size = 0.75) +
  geom_smooth(method = "lm") +
  facet_wrap(~ icgrp) +
  ggpubr::stat_cor(label.y.npc = 0.5) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  labs(y = "cg20786223",
       title = "Immune age-related CpG")

plot <- (a/b/c) + plot_annotation(tag_levels = "A")
ggsave(plot, file = "4-display-elements/panels/figs4.pdf", width = 4.75, height = 9)
```

\includegraphics[trim=0 0cm 0 0, width=0.6\textwidth]{./panels/figs4.pdf}

\begingroup \boldf Figure S4. Representative examples of CpG sites passing cell type-specific criteria. A \endgroup CpG correlated with chronological age in both epithelial (cervical and buccal samples with less than 20\% inferred immune cell proportion) and immune cells (blood), \textbf{B} only epithelial cells, or \textbf{c} only immune cells. Correlation coefficient is Pearson's rho.

\newpage


\invisiblesection{Figure S5. Idenfication of PCGT CpG sites in mice and age-related CpGs in BALB/c mice.}

```{r figs5, eval = F}
# A - PCGTs ------------------
# Load paper data and filter single occupancy
dat <- readxl::read_xls("0-data/files/mmc10.xls", skip = 1, col_types = "text")
dat <- dat[(dat$`Suz12 Occupancy`==1 | dat$`Eed Occupancy` == 1 | dat$`H3K27me3 Occupancy` == 1),]

# Get orthologues via Entrez
ortho <- dat$`EntrezGene ID` |> 
  babelgene::orthologs(species = 'mouse') |> 
  dplyr::mutate(`EntrezGene ID` = as.character(human_entrez)) |> 
  dplyr::rename(mgene = symbol)
dat <- dat |> 
  dplyr::left_join(dplyr::select(ortho, `EntrezGene ID`, mgene))

# groups:
# not on array
# on array
# on array and tss200
# on array, tss200 and fetal zero
load("3-mouse/3-output/pcgt.Rdata")
load("3-mouse/3-output/pcgt_cpgs_tss200_fetalzero.Rdata")
cpgs_single_tss200 <- cpgs_single_tss200 |> 
  data.frame() |> 
  dplyr::mutate(gene = stringr::str_split(GeneName_NCBI, ";", simplify = T)[,1])

pcgt_cpgs_tss200_fetalzero <- pcgt_cpgs_tss200_fetalzero |> 
  data.frame() |> 
  dplyr::mutate(gene = stringr::str_split(GeneName_NCBI, ";", simplify = T)[,1])

tmp <- dat |> 
  dplyr::filter(!is.na(mgene)) |> 
  dplyr::mutate(group = case_when(!mgene %in% pcgt_single ~ "not covered",
                                  mgene %in% pcgt_single & !mgene %in% c(cpgs_single_tss200$gene, pcgt_cpgs_tss200_fetalzero) ~ "covered",
                                  mgene %in% cpgs_single_tss200$gene & !mgene %in% pcgt_cpgs_tss200_fetalzero$gene ~ "covered (TSS200)",
                                  mgene %in% pcgt_cpgs_tss200_fetalzero$gene ~ "covered (TSS200, fetal unmethylated)"),
                group = factor(group,
                               levels = c("not covered", "covered", "covered (TSS200)", "covered (TSS200, fetal unmethylated)")))

a <- tmp |> 
  dplyr::group_by(group) |> 
  dplyr::count() |> 
  ggplot(aes(x = 1,
             y = n,
             fill = group),
         alpha = 0.6) +
  geom_col() +
  scale_fill_manual(values = cols1[c(1,3,5,7)],
                    name = "") +
  theme_minimal() +
  labs(x = "",
       y = "Unique PCGT gene orthologs") +
  theme(aspect.ratio = 4,
        axis.ticks.x = element_blank(),
        axis.text.x = element_blank())

# B - Propensity ------------------
liu_pred <- readxl::read_xls("0-data/files/prop-score-pcgt.xls", skip = 1) |> 
  janitor::clean_names() |> 
  dplyr::mutate(selected = ifelse(gene_name %in% pcgt_cpgs_tss200_fetalzero$GeneName_NCBI, "pcgt selected", "not selected"))


b <- liu_pred |> 
  ggplot(aes(x = 1, y = predicted_propensity_score, fill = selected)) +
  # clouds
  introdataviz::geom_flat_violin(trim=FALSE, alpha = 0.4,
                                 position = position_nudge(x = rain_height+.05)) +
  # boxplots
  geom_boxplot(width = 0.2, alpha = 0.4, show.legend = FALSE, 
               outlier.shape = NA) +
  stat_summary(fun.data = mean_cl_normal, mapping = aes(color = selected),
               show.legend = FALSE,
               position = position_nudge(x = rain_height * 3)) +
  scale_colour_manual(values = cols[c(3, 7)],
                      aesthetics = c('fill', 'colour'),
                      name = '',
                      labels = c("not selected for PCGTs",
                                 "selected for PCGTs")) +
  coord_flip() +
  theme_bw() +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.title.x = element_markdown(),
        legend.position = 'top') +
  labs(y = '<b>Predicted Propensity score*</b><br>(*Liu et al., 2010, <i>Genomics</i>)',
       x = '')

# C - Age association ------------------
load("3-mouse/2-output/age.Rdata")

c <- cors |> 
  ggplot(aes(x = pval)) +
  geom_histogram(bins = 200) +
  theme_minimal() +
  labs(x = "P value",
       y = "n") +
  theme_bw()+
  theme(aspect.ratio = 1)

# D - Overlaps ------------------
load("3-mouse/4-output/mPCGTgen_mPCGTnonage_annotated.Rdata")

library(ggvenn)
v <- list(all = tmp$mgene,
          mPCGTgen = unique(mPCGTgen$gene),
          mPCGTnonage = unique(mPCGTnonage$gene))

d <- ggvenn(v,
       fill_color = cols[c(2,4,6)],
       stroke_size = 0.5,
       text_size = 2.8,
       set_name_size = 3)

# E - example PCGTgen ------------------
load("0-data/dat_assessment_mouse.Rdata")

top2_age <- mPCGTgen |> arrange(pval) |>
  dplyr::slice(1:2)

tmp <- pheno |> 
  dplyr::filter(substudy == 'Age study') |> 
  tidyr::pivot_longer(any_of(c(top2_age$cg, mPCGTnonage[1:2,]$cg)),
                      names_to = 'cg',
                      values_to = 'beta') |> 
  dplyr::mutate(group = ifelse(cg %in% top2_age$cg, "age-associated", "non-age-associated"))

theme_set(
  theme_bw() +
  theme(legend.position = 'right')
)

e <- tmp |> 
  ggplot(aes(x = age,
             y = beta,
             colour = tissue)) +
  geom_point() +
  geom_smooth(method = 'lm',
              se = F) +
  facet_wrap(group~cg, nrow = 1) +
  ggpubr::stat_cor(size = 2.9) +
  scale_colour_manual(values = cols,
                      name = '') +
  labs(x = "age (days)",
       y = "methylation beta value") +
  theme(aspect.ratio = 1)

plot <- (a|b|c)/(d) +
  plot_annotation(tag_levels = 'a')

f <- pheno |> 
  dplyr::filter(substudy == 'Age study') |>
  tidyr::pivot_longer(c(idx_mPCGTgen, idx_mPCGTnonage),
                      names_to = 'idx',
                      values_to = 'value') |> 
  dplyr::mutate(idx = gsub("idx_", "", idx)) |> 
  ggplot(aes(x = age,
             y = value,
             colour = tissue)) +
  geom_point() +
  geom_smooth(method = 'lm',
              se = F) +
  facet_wrap(~idx, nrow = 1) +
  ggpubr::stat_cor(size = 2.9) +
  scale_colour_manual(values = cols,
                      name = '') +
  labs(x = "age (days)",
       y = "methylation beta value") +
  theme(aspect.ratio = 1)

plot <- (a|b)/(c|d)/(e)/(f) +
  plot_annotation(tag_levels = 'A')

cairo_pdf("4-display-elements/panels/figs5.pdf",width = 12, height = 15)
print(plot)
dev.off()
```

\includegraphics{./panels/figs5.pdf}


\begingroup \boldf Figure S5. Idenfication of PCGT sites and age-related CpGs in BALB/c mice. A \endgroup Unique murine PCGT orthologs and their coverage on the mouse methylation array. \begingroup \boldf B \endgroup Plot of the Liu \textit{et al.} propensity score of genes that were selected as PCGT genes or not selected as PCGT genes in the current study. \begingroup \boldf C \endgroup P value histogram for association of age with beta methylation levels after correcting for tissue type. \begingroup \boldf D \endgroup Overlap of all unique PCGT orthologs covered on the mouse methylation array with sites covered in age-related mouse PCGTgen (mPCGTgen) or non-age-related mPCGTnonage sites. \begingroup \boldf E \endgroup Methylation levels in various tissues in example sites for age-associated or non-age-associated CpGs covered in mPCGTgen and mPCGTnonage, respectively. \begingroup \boldf F \endgroup Mean methylation levels in various tissues for mPCGTgen and mPCGTnonage index.

\newpage

\invisiblesection{Figure S6. Principal component analysis of mouse tissue DNAm data.}

```{r figs6, eval = F}
load('0-data/dat_assessment_mouse.Rdata')

p <- pheno |> 
  dplyr::filter(substudy == 'Age study')|> 
  ggplot(aes(x = PC1,
             y = PC2,
             colour = tissue)) +
  geom_point(size = 2.5,
             alpha = 0.8) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_colour_manual(values = colorRampPalette(cols)(10))

cairo_pdf("4-display-elements/panels/figs6.pdf",width = 7, height = 4.5)
print(p)
dev.off()
```

\includegraphics{./panels/figs6.pdf}

\begingroup \boldf Figure S6. Principal component analysis of mouse tissue DNAm data. \endgroup The first two principal components of the top 30,000 variable CpGs are shown for the age study.

\newpage 

\invisiblesection{Table S1. Datasets and accession numbers used in this study.}

```{r s.table1}
load(here("3-evaluation/3-output/data.Rdata"))

s1 <- data |> 
  dplyr::filter(!(group == "OSK reprogramming" & accession == "NA") & accession != "GSE147436") |> 
  group_by(group, celltype, accession) |>
  reframe(n_control = sum(type %in% c("Control", "Never smoker", "Control (WT)", "Control (BRCA)")),
          n_case = sum(!type %in% c("Control", "Never smoker", "Control (WT)", "Control (BRCA)"))) |> 
  arrange(group, celltype, accession) |> 
  ungroup() |> 
  dplyr::mutate(repository = case_when(grepl("GSE", accession) ~ "NCBI GEO",
                                       grepl("EGAS", accession) ~ "EGA",
                                       grepl("TCGA", accession) ~ "TCGA",
                                       grepl("NA", accession) ~ "NA"),
                celltype = gsub("breast tissue", "breast", celltype)) |> 
  dplyr::add_row(group = 'proliferation',
                 repository = 'NCBI GEO',
                 celltype = 'iliac vein',
                 accession = 'GSE197512',
                 n_control = 12,
                 n_case = 12) |> 
  dplyr::add_row(group = 'proliferation',
                 repository = 'NCBI GEO',
                 celltype = 'skin, arm',
                 accession = 'GSE197512',
                 n_control = 6,
                 n_case = 6) |> 
  dplyr::add_row(group = 'mouse methylation',
                 repository = 'NCBI GEO',
                 celltype = 'mammary gland',
                 accession = 'GSE236260',
                 n_control = 18,
                 n_case = 17) |> 
  dplyr::add_row(group = 'mouse methylation',
                 repository = 'NCBI GEO',
                 celltype = 'cervix',
                 accession = 'GSE236260',
                 n_control = 18,
                 n_case = 17) |> 
  dplyr::add_row(group = 'mouse methylation',
                 repository = 'NCBI GEO',
                 celltype = 'PBMCs',
                 accession = 'GSE236260',
                 n_control = 18,
                 n_case = 17) |> 
  dplyr::select(group, repository, celltype, accession, n_control, n_case) |> 
  dplyr::mutate(stage = case_when(group == 'proliferation' & celltype == 'skin, sacrum' ~ 'discovery (proliferation assocation)/validation',
                                  group == 'proliferation' & celltype != 'skin, sacrum' ~ 'discovery (proliferation association)',
                                  group == 'senescence' ~ 'discovery (senescence association)/validation',
                                  group == 'current cancer;surrogate tissue' & celltype == 'cervical sample' ~ 'discovery (age correlation, controls only)/validation',
                                  TRUE ~ 'validation')) |> 
  write.table(file = here("0-data/Table-S1.txt"),
              sep = "\t", quote = F, row.names = F)
```

\begingroup \boldf Table S1. Datasets and accession numbers used in this study. \endgroup See .txt for Supplementary Table 1.

\invisiblesection{Table S2. Overview of CpGs in age signatures defined in the current study.}

```{r, eval = F}
coefs <- list.files("3-evaluation/1-output", full.names = T, pattern = "*.Rds")
coefnames <- gsub(".Rds", "", basename(coefs))
out <- vector(mode = "list", length = length(coefs))
names(out) <- paste0("idx_", coefnames)
  
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

for (i in 1:length(out)){
  
  cat(i, "/ 22\n")
  
    wx <- readRDS(coefs[i])
    index = names(out)[i]
    
    tmp <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19, lociNames = names(wx))) |> 
      dplyr::select(chr, pos, Name, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Group) |> 
      dplyr::mutate(hg = 'hg19',
                    clock = gsub("idx_", "", names(out)[i])) |> 
      dplyr::relocate(Name)
    
    tmp$weight_clock <- wx[match(tmp$Name, names(wx))]
    rownames(tmp) <- NULL
    
    if(i == 1){
      dat <- tmp
    } else {
      dat <- rbind(dat, tmp)
    }
}

write.table(dat, file = "0-data/Table-S2.csv", quote = F, sep = ",", row.names = F)

```

\begingroup \boldf Table S2. Overview of CpGs in age signatures defined in the current study. \endgroup See .csv file for Table S2.

\invisiblesection{Table S3. CpGs associated with senescence in the current study.}

```{r, eval = F}
load("1-functional-association/1-senescence/3-output/coef.Rdata")

library(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)

# Remove any chrX/Y CpGs
anno_epic <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
anno_450k <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
chrX_Y <- unique(c(anno_epic[anno_epic$chr%in% c("chrX", "chrY"),]$Name,
                   anno_450k[anno_450k$chr%in% c("chrX", "chrY"),]$Name))

sen <- df |> 
  dplyr::filter(!cg %in% chrX_Y) |> 
  dplyr::mutate(padj = p.adjust(p, method = "fdr")) |> 
  dplyr::filter(padj < 0.05)

tmp <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19, lociNames = sen$cg)) |> 
      dplyr::select(chr, pos, Name, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Group) |> 
      dplyr::mutate(hg = 'hg19') |> 
      dplyr::relocate(Name)

sen <- sen[match(tmp$Name, sen$cg),]

tmp$coef <- sen$coef
tmp$pval <- sen$p
tmp$pval_adj <- sen$padj    
rownames(tmp) <- NULL

write.table(tmp, file = "0-data/Table-S3.csv", quote = F, sep = ",", row.names = F)
```

\begingroup \boldf Table S3. CpGs associated with senescence in the current study. \endgroup See .csv file for Table S3.

\invisiblesection{Table S4. CpGs associated with proliferation in the current study.}

```{r, eval = F}
load("1-functional-association/2-proliferation/3-output/coef.Rdata")

library(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)

# Remove any chrX/Y CpGs
anno_epic <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
anno_450k <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
chrX_Y <- unique(c(anno_epic[anno_epic$chr%in% c("chrX", "chrY"),]$Name,
                   anno_450k[anno_450k$chr%in% c("chrX", "chrY"),]$Name))

pro <- df |> 
  dplyr::filter(!cg %in% chrX_Y) |> 
  dplyr::mutate(padj = p.adjust(p, method = "fdr")) |> 
  dplyr::filter(padj < 0.05) |> 
  # invert coefficient so direction is more intuitive
  dplyr::mutate(coef = coef*-1)

tmp <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19, lociNames = pro$cg)) |> 
      dplyr::select(chr, pos, Name, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Group) |> 
      dplyr::mutate(hg = 'hg19') |> 
      dplyr::relocate(Name)

pro <- pro[match(tmp$Name, pro$cg),]

tmp$coef <- pro$coef
tmp$pval <- pro$p
tmp$pval_adj <- pro$padj    
rownames(tmp) <- NULL

write.table(tmp, file = "0-data/Table-S4.csv", quote = F, sep = ",", row.names = F)
```

\begingroup \boldf Table S4. CpGs associated with proliferation in the current study. \endgroup See .csv file for Table S4.

\invisiblesection{Table S5. PCGT-associated CpGs in the current study.}

```{r, eval = F}
load("1-functional-association/3-pcgt/1-output/pcgt_cpgs_tss200_fetalzero.Rdata")

library(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)

# Remove any chrX/Y CpGs
anno_epic <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
anno_450k <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
chrX_Y <- unique(c(anno_epic[anno_epic$chr%in% c("chrX", "chrY"),]$Name,
                   anno_450k[anno_450k$chr%in% c("chrX", "chrY"),]$Name))

pcgt <- pcgt_cpgs_tss200_fetalzero |> 
  as.data.frame() |> 
  dplyr::filter(!Name %in% chrX_Y)

tmp <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b2.hg19, lociNames = pcgt$Name)) |> 
      dplyr::select(chr, pos, Name, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Group) |> 
      dplyr::mutate(hg = 'hg19') |> 
      dplyr::relocate(Name)
rownames(tmp) <- NULL

write.table(tmp, file = "0-data/Table-S5.csv", quote = F, sep = ",", row.names = F)
```

\begingroup \boldf Table S5. PCGT-associated CpGs in the current study. \endgroup See .csv file for Table S5.

\invisiblesection{Table S6. Reference and comparison groups in Figure 3.}

\begingroup \boldf Table S7. Reference and comparison groups in Figure 3. \endgroup See .xlsx for Table S6.

\invisiblesection{Table S7. CpG Overlap criteria.}

\begingroup \boldf Table S7. CpG Overlap criteria. \endgroup See .xlsx for Table S7.




